To: vim_dev@googlegroups.com
Subject: Patch 7.3.041
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.041
Problem:    Compiler warning for accessing mediumVersion. (Tony Mechelynck)
Solution:   Use the pointer instead of the array itself. (Dominique Pelle)
Files:	    src/version.c


*** ../vim-7.3.040/src/version.c	2010-10-27 13:37:39.000000000 +0200
--- src/version.c	2010-10-27 15:58:26.000000000 +0200
***************
*** 1342,1350 ****
  	if (highest_patch())
  	{
  	    /* Check for 9.9x or 9.9xx, alpha/beta version */
! 	    if (isalpha((int)mediumVersion[3]))
  	    {
! 		if (isalpha((int)mediumVersion[4]))
  		    sprintf((char *)vers + 5, ".%d%s", highest_patch(),
  							   mediumVersion + 5);
  		else
--- 1344,1352 ----
  	if (highest_patch())
  	{
  	    /* Check for 9.9x or 9.9xx, alpha/beta version */
! 	    if (isalpha((int)vers[3]))
  	    {
! 		if (isalpha((int)vers[4]))
  		    sprintf((char *)vers + 5, ".%d%s", highest_patch(),
  							   mediumVersion + 5);
  		else
*** ../vim-7.3.040/src/version.c	2010-10-27 13:37:39.000000000 +0200
--- src/version.c	2010-10-27 15:58:26.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     41,
  /**/

-- 
ARTHUR:        I command you as King of the Britons to stand aside!
BLACK KNIGHT:  I move for no man.
                                  The Quest for the Holy Grail (Monty Python)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\        download, build and distribute -- http://www.A-A-P.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
